src/server/game/AuctionHouse/AuctionHouseMgr.cpp

```diff
@@ -648,105 +648,128 @@ void AuctionHouseObject::BuildListOwnerItems(WorldPacket& data, Player* player,
 void AuctionHouseObject::BuildListAuctionItems(WorldPacket& data, Player* player,
     std::wstring const& wsearchedname, uint32 listfrom, uint8 levelmin, uint8 levelmax, uint8 usable,
     uint32 inventoryType, uint32 itemClass, uint32 itemSubClass, uint32 quality,
-    uint32& count, uint32& totalcount)
+    uint32& count, uint32& totalcount, bool getall)
 {
     int loc_idx = player->GetSession()->GetSessionDbLocaleIndex();
     int locdbc_idx = player->GetSession()->GetSessionDbcLocale();
 
     time_t curTime = sWorld->GetGameTime();
+    time_t throttleTime = GetAllThrottleMap.find(player->GetGUID()) != GetAllThrottleMap.end() ? GetAllThrottleMap[player->GetGUID()] : curTime;
 
-    for (AuctionEntryMap::const_iterator itr = AuctionsMap.begin(); itr != AuctionsMap.end(); ++itr)
+    if (getall && throttleTime <= curTime)
     {
-        AuctionEntry* Aentry = itr->second;
-        // Skip expired auctions
-        if (Aentry->expire_time < curTime)
-            continue;
-
-        Item* item = sAuctionMgr->GetAItem(Aentry->itemGUIDLow);
-        if (!item)
-            continue;
+        for (AuctionEntryMap::const_iterator itr = AuctionsMap.begin(); itr != AuctionsMap.end(); ++itr)
+        {
+            AuctionEntry* Aentry = itr->second;
+            // Skip expired auctions
+            if (Aentry->expire_time < curTime)
+                continue;
 
-        ItemTemplate const* proto = item->GetTemplate();
+            Item* item = sAuctionMgr->GetAItem(Aentry->itemGUIDLow);
+            if (!item)
+                continue;
 
-        if (itemClass != 0xffffffff && proto->Class != itemClass)
-            continue;
+            ++count;
+            ++totalcount;
+            Aentry->BuildAuctionInfo(data);
+        }
+        GetAllThrottleMap[player->GetGUID()] = curTime + sWorld->getIntConfig(CONFIG_AUCTION_GETALL_DELAY);
```

Comment:
Perhaps add a return after this statement so the 100 lines after this one do not need to be moved to the `else` (it makes code reviewing harder)